Sources/PackageCollectionsSigning/Certificate/CertificatePolicy.swift

```diff
@@ -18,9 +30,362 @@ protocol CertificatePolicy {
+            return Data([0x2B, 0x06, 0x01, 0x05, 0x05, 0x07, 0x03, 0x03])
+        }
+    }
+
+    #else
+    var flag: CInt {
+        switch self {
+        case .codeSigning:
+            // https://www.openssl.org/docs/man1.1.0/man3/X509_get_extension_flags.html
+            return XKU_CODE_SIGN
+        }
     }
+    #endif
+}
+
+extension CertificatePolicy {
+    static func loadCerts(at directory: URL) -> [Certificate] {
+        var certs = [Certificate]()
+        if let enumerator = FileManager.default.enumerator(at: directory, includingPropertiesForKeys: nil) {
+            for case let fileURL as URL in enumerator {
+                do {
+                    certs.append(try Certificate(derEncoded: Data(contentsOf: fileURL)))
+                } catch {
+                    // Skip cert if problematic
+                }
+            }
+        }
+        return certs
+    }
+}
+
+enum CertificatePolicyError: Error {
+    case emptyCertChain
+    case certVerificationFailure
+    case extensionFailure
+//    case ocspFailure
+}
+
+// MARK: - Certificate policies
+
+/// Default policy for validating certificates used to sign package collections.
+///
+/// Certificates must satisfy these conditions:
+///   - The timestamp at which signing/verification is done must fall within the signing certificate’s validity period.
+///   - The certificate’s “Extended Key Usage” extension must include “Code Signing”.
+///   - The certificate must use either 256-bit EC (recommended) or 2048-bit RSA key.
+///   - The certificate must not be revoked. The certificate authority must support OCSP, which means the certificate must have the
+///   "Certificate Authority Information Access" extension that includes OCSP as a method, specifying the responder’s URL.
+///   - The certificate chain is valid and root certificate must be trusted.
+struct DefaultCertificatePolicy: CertificatePolicy {
+    let trustedRoots: [Certificate]?
+    let expectedSubjectUserID: String?
+
+    let queue: DispatchQueue
+
+    /// Initializes a `DefaultCertificatePolicy`.
+    /// - Parameters:
+    ///   - trustedRootCertsDir: On Apple platforms, all root certificates that come preinstalled with the OS are automatically trusted.
+    ///                          Users may specify additional certificates to trust by placing them in `trustedRootCertsDir` and
+    ///                          configure the signing tool or SwiftPM to use it. On non-Apple platforms, only trust root certificates in
+    ///                          `trustedRootCertsDir` are trusted.
+    ///   - expectedSubjectUserID: The subject user ID that must match if specified.
+    ///   - queue: The `DispatchQueue` to perform async operations.
+    init(trustedRootCertsDir: URL? = nil, expectedSubjectUserID: String? = nil, queue: DispatchQueue = DispatchQueue.global()) {
+        self.trustedRoots = trustedRootCertsDir.map { Self.loadCerts(at: $0) }
```

Comment:
is loading them quick? otherwise maybe you can load and cache in "validate" sync its async anyways?